### DISCUSSION TITLE
How to reduce aot file size

### DISCUSSION THREAD

**Discussion by howzew** [AvaloniaUI/Avalonia#14219] [Q&A] [answered]
I created an avalonia mvvm project without adding any new code. After aot published, the file size reached 21M.
I have used PublishTrimmed, and there are three files in the directory, "av_libglesv2", "libHarfBuzzSharp", and "libSkiaSharp". 
Such a blank project actually has 21M. How should I reduce this size?

**Comment by maxkatz6** [comment]
You can get smaller app by removing Avalonia.Fonts.Inter package.
But in general, it's pretty much what you can get in .NET normally. App size won't grow as much with more code you add. Though you need to be careful with other dependencies or assets.

**Reply by howzew** [reply]
I deleted the Avalonia.Fonts.Inter package and the program reduced the size by 3M. Also, were those three files generated by using avalonia? Because I used the console to test aot, it seems that these three files do not exist. Can they be packaged into the files after aot?

**Reply by maxkatz6** [reply, marked as answer]
Yes, these files are required as it's a native Skia libraries and dependencies.

You can try to static link them following this thread: https://github.com/AvaloniaUI/Avalonia/issues/9503
But usually, it's not worth it.

**Reply by howzew** [reply]
Thank you

**Comment by sgf** [comment]
[Is it possible for Avalonia to train an AI to convert libraries such as av libglesv2, libHarfBuzzSharp, and libSkiaSharp into efficient pure C# implementations?](https://github.com/AvaloniaUI/Avalonia/discussions/15818)
